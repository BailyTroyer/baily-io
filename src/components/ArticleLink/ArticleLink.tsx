import React, { FC } from "react"

import { useStaticQuery, graphql } from "gatsby"
import Img from "gatsby-image"

interface Props {
  tag: string
  title: string
  date: string
}

const ArticleLink: FC<Props> = (props: Props) => {
  const { tag, title, date } = props

  const data = useStaticQuery(graphql`
    query {
      cloudImage: file(relativePath: { eq: "cloud.png" }) {
        childImageSharp {
          fluid(maxWidth: 100) {
            ...GatsbyImageSharpFluid
          }
        }
      }
    }
  `)

  return (
    <div className="my-6 flex flex-row items-center cursor-pointer">
      <Img
        fluid={data.cloudImage.childImageSharp.fluid}
        className="hidden md:flex rounded-lg object-contain w-20 h-20 mr-4"
      />
      <div>
        <p className="font-bold text-purple-700 text-xs">{tag.toUpperCase()}</p>
        <h2 className="font-extrabold text-gray-900 text-xl">{title}</h2>
        <p className="font-medium text-gray-900 text-xs">{date}</p>
      </div>
    </div>
  )
}

export default ArticleLink
