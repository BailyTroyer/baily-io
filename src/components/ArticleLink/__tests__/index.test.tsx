import React from "react"

import * as Gatsby from "gatsby"
import renderer from "react-test-renderer"

import ArticleLink from "../ArticleLink"

const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery")
useStaticQuery.mockImplementation(() => ({
  cloudImage: {
    childImageSharp: {
      fluid: {
        aspectRatio: 0,
        src: "",
        srcSet: "",
        sizes: "",
      },
    },
  },
}))

it("renders correctly", () => {
  const tree = renderer
    .create(<ArticleLink tag="TAG" title="TITLE" date="DATE" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
