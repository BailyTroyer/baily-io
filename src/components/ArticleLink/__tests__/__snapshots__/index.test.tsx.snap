// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="my-6 flex flex-row items-center cursor-pointer"
>
  <div
    className="hidden md:flex rounded-lg object-contain w-20 h-20 mr-4 gatsby-image-wrapper"
    style={
      Object {
        "maxHeight": null,
        "maxWidth": null,
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      aria-hidden={true}
      style={
        Object {
          "paddingBottom": "Infinity%",
          "width": "100%",
        }
      }
    />
    <picture>
      
      <img
        alt=""
        loading="lazy"
        onLoad={[Function]}
        sizes=""
        src=""
        srcSet=""
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 500ms",
            "width": "100%",
          }
        }
      />
    </picture>
    <noscript
      dangerouslySetInnerHTML={
        Object {
          "__html": "<picture><source srcset=\\"\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
        }
      }
    />
  </div>
  <div>
    <p
      className="font-bold text-purple-700 text-xs"
    >
      TAG
    </p>
    <h2
      className="font-extrabold text-gray-900 text-xl"
    >
      TITLE
    </h2>
    <p
      className="font-medium text-gray-900 text-xs"
    >
      DATE
    </p>
  </div>
</div>
`;
